tooling/minitest/src/tests/bit_and.rs

```diff
@@ -0,0 +1,96 @@
+use crate::*;
+
+/// Test that bit_and works for ints
+#[test]
+fn bit_and_int_works() {
+    let locals = [<i32>::get_type()];
+
+    let b0 = block!(
+        storage_live(0),
+        // test all +- combinations
+        assign(local(0), bit_and::<i32>(const_int::<i32>(171), const_int::<i32>(62),)),
+        assign(local(0), bit_and::<i32>(const_int::<i32>(-171), const_int::<i32>(62),)),
+        assign(local(0), bit_and::<i32>(const_int::<i32>(171), const_int::<i32>(-62),)),
+        assign(local(0), bit_and::<i32>(const_int::<i32>(-171), const_int::<i32>(-62),)),
+        storage_dead(0),
+        exit(),
+    );
+
+    let prog = program(&[function(Ret::No, 0, &locals, &[b0])]);
+    assert_stop(prog);
+}
+
+/// Test that bit_and works for ints
+#[test]
+fn bit_and_bool_works() {
+    let locals = [<bool>::get_type()];
+    let b0 = block!(
+        storage_live(0),
+        assign(local(0), bit_and::<bool>(const_bool(false), const_bool(false),)),
+        assign(local(0), bit_and::<bool>(const_bool(false), const_bool(true),)),
+        assign(local(0), bit_and::<bool>(const_bool(true), const_bool(false),)),
+        assign(local(0), bit_and::<bool>(const_bool(true), const_bool(true),)),
```

Comment:
This will not check that the correct value was actually computed, will it?

The new tests for bools should be in `bool.rs`, and for the ints we probably want an `int.rs` to put the tests there.

Reply:
No it doesn't. So far `minitest` `run_program` does not expose any computed values.

 I split the code into `bool.rs` and `int.rs`.

Reply:
Yeah but you can add an `if` in the code that calls `unreachable` if anything goes wrong, and then the test would fail if we ever hit the `unreachable`.

There should be existing tests that demonstrate this technique.

Reply:
See for instance here:

https://github.com/minirust/minirust/blob/f4098857543ac6f72d3c441a597dbd7e12736e75/tooling/minitest/src/tests/enum_discriminant.rs#L88-L101

Reply:
Ah yes that is quite neat.